aalc: modify flow reading to average

khoung76642 opened the pull request:
Summary:
- fix GPIO definition of board stage
- modify flow reading to average
- fix stop_pump not working when there is no mode
- modify log data of pump/hex fan to the set value
- add get board stage modbus command
- add get fan table revision modbus command
- modify RPU level threshold retry times to 5
- modify flow rate threshold retry times to 20

Test Plan:
- Build code: PASS

waffle2k commented:
Hi there -- please address the cpp check linter before we're able to accept the change

khoung76642 commented:
Hi, We fix the problem of cpp check, please help us to merge this branch.
Thanks！

This change will modify zephyr code, so we did't modify this item. 
(meta-facebook/aalc-rpu/src/platform/plat_modbus.c:1463:76:Parameter 'reg' can be declared as pointer to const. However it seems that 'holding_reg_multi_wr' is a callback function, if 'reg' is declared with const you might also need to cast function pointer(s).)

jamesatha commented on meta-facebook/aalc-rpu/src/platform/plat_class.h:
```diff
@@ -33,6 +33,8 @@ enum TEMP_MODULE {
 enum BOARD_STAGE {
 	BOARD_STAGE_EVT,
 	BOARD_STAGE_DVT,
+	BOARD_STAGE_PVT, // no this stage
```
fix comment

khoung76642 replied:
we fixed the comment.

jamesatha commented on meta-facebook/aalc-rpu/src/platform/plat_hook.c:
```diff
@@ -1024,6 +1026,41 @@ bool post_adm1272_read(sensor_cfg *cfg, void *args, int *reading)
+static double averge_flow_rate(double val)
+{
+	bool is_cache = false;
+	for (uint8_t i = 0; i < FLOW_SENSOR_CACHE_MAX_NUM; i++) {
```
This loop seems inefficient. Let's create an `index` and `isFull` variables. That way, we don't have to do this scan every time. Additionally, flow_cache_data can become just an array of `float`s

khoung76642 replied:
We created the "flow_cache_latest_index" variable as a pointer and the "flow_cache_is_full" variable as a flag, then we make this function more efficient.
Also, we changed some value type from double to float.

jamesatha commented on meta-facebook/aalc-rpu/src/platform/plat_hook.c:
```diff
@@ -1024,6 +1026,41 @@ bool post_adm1272_read(sensor_cfg *cfg, void *args, int *reading)
+		}
+	}
+
+	return (count_num > 0) ? (total_val / count_num) : 0.0;
```
You just added a value to the list, so count_num can never be <= 0

khoung76642 replied:
We modified this part that will return (total_val / count_num) without checking value is <0 or not.

khoung76642 commented:
Hi, we have modified the above suggestions and rebased on the newest main branch. If there are no further questions, please help us merge this branch.

Thanks!
